### PR TITLE
Aluminum revolution

### PR DESCRIPTION
**Pull request by Karol1223**
#### Summary
Content "Adds new ways to obtain aluminum scrap as well as multiple new items made from aluminum."

#### Purpose of change
I added aluminum scrap a few weeks back and its only use was for crafting bronze, which felt stupid to say the least. This PR not only expands the range of items that give aluminum scrap when disassembled, it also adds several new items made from aluminum and adds scrap aluminum as a viable ingredient for most recipes using aluminum ingots. Also adds a new variant of makeshift scissors, so that aluminum scissors are no longer crafted from steel.

#### Describe the solution
All this PR does right now includes:
- Adds disassembly recipe for the expandable baton and golf club
- Modifies disassembly recipe for a range of items with aluminum as a material in most cases swapping scrap metal to scrap aluminum
- Adds several new items (two sizes of aluminum drums, aluminum boat hull, aluminum pot and frying pan, aluminum stepladder, makeshift scissors)
- Finally makes aluminum foil spawn in homes
- Modifies most recipes with aluminum ingots as ingredients to also use scrap aluminum
- Adds a recipe to melt down scrap aluminum into aluminum ingots and aluminum foil into scrap aluminum as well as a recipe to mix aluminum cans in order to create aluminum scrap
- Lowers the aluminum ingot requirement in crafting extra-light frames to reflect the product item's weight more accordingly
- Makes the current scissor recipe give 'makeshift scissors' instead, a new recipe to make regular scissors has been added at a slightly higher fabrication level

#### Describe alternatives you've considered
First of all I considered not doing all of this because it sparked as a foundation for a different PR I wanted to make, one that would change recipes for crafted electronics (with aluminum as the material) to use aluminum scrap instead of scrap metal. I was, however, suggested to include more aluminum sources while I'm at it and this kind of spiraled out of control. I also considered adding more aluminum-based items (foldable extra-light frames and candy wrappers, primarily) but didn't want to bloat the PR's size even more. I briefly thought about adding more 'furniture' variety i.e. aluminum windowframes and mailboxes, however with my understanding of how mapgen works this wouldn't be viable without touching mapgen itself and I am not going anywhere near that. I was also suggested to make boats that utilise the new aluminum frames actually spawn but I was afraid to attempt this.

#### Testing
Disassembled all the items that are supposed to have modified/added disassembly recipes. Tested and checked that new/modified crafting recipes do work as intended. Created a new world and teleported around kitchens to check that aluminum foil and new pots and pans did indeed spawn.

#### Additional context
I am open to expanding on this PR, however, I believe this should be done in separate PRs altogether, this one's kind of big already. I still intend to modify the crafting recipes for electronics after this is merged and I am seriously tempted to actually try adding the extra-light folding frames.

### CODE REVIEW

File: data/json/items/vehicle/boat.json [hunk 1]
```diff
@@ -125,5 +125,21 @@
...
     "to_hit": -1,
     "category": "veh_parts"
+  },
+  {
+    "type": "GENERIC",
+    "id": "aluminum_boat_hull",
+    "symbol": "o",
+    "color": "light_gray",
+    "name": { "str": "aluminum boat hull" },
+    "description": "An aluminum hull that keeps the boat afloat.  Add boat hulls to a vehicle until it floats.  Then attach oars or a motor to get the boat to move.",
```
Comment (TheShadowFerret) [inline data/json/items/vehicle/boat.json:135]:
```suggestion
    "description": "An aluminum hull that keeps the boat afloat.  Add boat hulls to a vehicle until it floats, then attach oars or a motor to get the boat to move.",
```

File: data/json/items/tool/misc.json [hunk 2]
```diff
@@ -785,5 +785,22 @@
...
     ],
     "flags": [ "FIRESTARTER", "THERMOMETER" ]
+  },
+  {
+    "id": "aluminum_stepladder",
+    "type": "TOOL",
+    "name": { "str": "aluminum stepladder" },
+    "description": "This is an aluminum stepladder.  Use it to set it down.",
+    "looks_like": "stepladder",
+    "weight": "91500 g",
```
Comment (AtomicFox556) [inline data/json/items/tool/misc.json:795]: I think an aluminum stepladder should be lighter than a wooden one.

https://www.laddersandscaffoldtowers.co.uk/blog/ladders/wooden-ladder-vs-aluminium-ladders/

Comment (Karol1223) [reply data/json/items/tool/misc.json:795]: Not gonna lie, I thought so as well. This mass is the result of me doing proportion calculations between the volume-mass ratio of planks and the wooden stepladder to calculate what the aluminum one should be but either the method is incredibly flawed or I messed up the maths. I'll go with the average weight of a 6-foot aluminum stepladder here

**Comment by Karol1223** [comment]
> Probably should have spawns for the aluminum barrels and boat hull

For the boat hull this would require making a new vehicle, wouldn't it? It is not something I'm confident enough to attempt. When it comes to the barrels I _could_ give them spawns, they should probably have a chance to spawn in place of steel ones if anything

**Comment by bombasticSlacks** [comment]
> they should probably have a chance to spawn in place of steel ones if anything

I think that is reasonable enough

File: data/json/recipes/other/parts.json [hunk 3]
```diff
@@ -384,7 +384,7 @@
       { "proficiency": "prof_welding_basic", "fail_multiplier": 1.5 },
       { "proficiency": "prof_welding" }
     ],
-    "components": [ [ [ "material_aluminium_ingot", 8 ] ] ]
+    "components": [ [ [ "material_aluminium_ingot", 8 ], [ "scrap_aluminum", 35 ] ] ]
```
Comment (Maleclypse) [inline data/json/recipes/other/parts.json:387]: Double checking that 8 ingots is convertible to 35 scrap?   How many scrap to make an ingot?

File: data/json/recipes/tools/containers.json [hunk 4]
```diff
@@ -1104,5 +1104,35 @@
...
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_CONTAINERS",
+    "skill_used": "fabrication",
+    "difficulty": 8,
+    "time": "7 h",
+    "autolearn": true,
+    "using": [ [ "blacksmithing_standard", 64 ] ],
+    "proficiencies": [ { "proficiency": "prof_metalworking" } ],
+    "qualities": [ { "id": "CHISEL", "level": 3 } ],
+    "components": [ [ [ "scrap_aluminum", 140 ], [ "material_aluminium_ingot", 12 ] ] ]
```
Comment (Maleclypse) [inline data/json/recipes/tools/containers.json:1121]: Different conversion rate used here than on my above question.

File: data/json/recipes/tools/containers.json [hunk 5]
```diff
@@ -1104,5 +1104,35 @@
...
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_CONTAINERS",
+    "skill_used": "fabrication",
+    "difficulty": 8,
+    "time": "4 h 20 m",
+    "autolearn": true,
+    "using": [ [ "blacksmithing_standard", 40 ] ],
+    "proficiencies": [ { "proficiency": "prof_metalworking" } ],
+    "qualities": [ { "id": "CHISEL", "level": 3 } ],
+    "components": [ [ [ "scrap_aluminum", 84 ], [ "material_aluminium_ingot", 7 ] ] ]
```
Comment (Maleclypse) [inline data/json/recipes/tools/containers.json:1136]: And a third different conversion rate it looks like.

**Review by Maleclypse — COMMENTED** [review]
From going over this PR it looks like you need a stable conversion amount between scrap and ingots and I don't believe I saw a recipe that converts ingots into scrap but if I missed it please direct me to where I can see it.

**Comment by Karol1223** [comment]
> From going over this PR it looks like you need a stable conversion amount between scrap and ingots and I don't believe I saw a recipe that converts ingots into scrap but if I missed it please direct me to where I can see it.

I didn't use a conversion rate between ingots and scrap, instead simply checked to make sure the mass stays vaguely the same. A 200l aluminum drum weighs 7000g so it's either 12 625g ingots (technically 11,2; using 11 wouldn't make sense) or 140 50g scrap. This is the rule I kept throughout all the recipes utilising both of them and also the reason why I lowered the amount of ingots needed to craft aluminum frames

**Comment by Maleclypse** [comment]
> > From going over this PR it looks like you need a stable conversion amount between scrap and ingots and I don't believe I saw a recipe that converts ingots into scrap but if I missed it please direct me to where I can see it.
> 
> I didn't use a conversion rate between ingots and scrap, instead simply checked to make sure the mass stays vaguely the same. A 200l aluminum drum weighs 7000g so it's either 12 625g ingots (technically 11,2 but using 11 wouldn't make sense) or 140 50g scrap. This is the rule I kept throughout all the recipes utilising both of them and also the reason why I lowered the amount of ingots needed to craft aluminum frames
> 
> Having said that if you believe the method to calculate this should be different I could change it but this seemed like the most reasonable way to calculate it

Then you'd need to change this one `"components": [ [ [ "material_aluminium_ingot", 8 ], [ "scrap_aluminum", 35 ] ] ]` because 8 ingots would equal 96 scrap not 35.   So one of these items is wrong.

**Comment by Karol1223** [comment]
> Then you'd need to change this one `"components": [ [ [ "material_aluminium_ingot", 8 ], [ "scrap_aluminum", 35 ] ] ]` because 8 ingots would equal 96 scrap not 35. So one of these items is wrong.

Aaaand that was an extremely stupid mistake. In my local copy of the game I changed it, I lowered the aluminum ingot requirement to 4. I forgot to apply this change in the PR. So yes, now the aluminum frame _actually does_ use less aluminum ingots that it used to. I intended to make this change and I thought I did, turns out I overlooked it. My bad entirely

**Review by Maleclypse — APPROVED** [review]

**Pull request merged**